### PR TITLE
Update azure_firewall_modified_or_deleted.yml

### PR DESCRIPTION
**Pull request by TheLawsOfChaos**
Added sub-tech reference, new tactic, and sub-tech.

### CODE REVIEW

File: rules/cloud/azure/azure_firewall_modified_or_deleted.yml [hunk 1]
```diff
@@ -4,11 +4,14 @@ status: experimental
...
 references:
     - https://docs.microsoft.com/en-us/azure/role-based-access-control/resource-provider-operations
+    - https://attack.mitre.org/techniques/T1562/004/
 author: Austin Songer @austinsonger
 date: 2021/08/08
 modified: 2022/08/23
 tags:
     - attack.impact
+    - attack.defense_evasion
+    - attack.1562.004
```
Comment (nasbench) [inline rules/cloud/azure/azure_firewall_modified_or_deleted.yml:14]:
```suggestion
    - attack.t1562.004
```

File: rules/cloud/azure/azure_firewall_modified_or_deleted.yml [hunk 2]
```diff
@@ -4,11 +4,14 @@ status: experimental
 description: Identifies when a firewall is created, modified, or deleted.
 references:
     - https://docs.microsoft.com/en-us/azure/role-based-access-control/resource-provider-operations
+    - https://attack.mitre.org/techniques/T1562/004/
```
Comment (nasbench) [inline rules/cloud/azure/azure_firewall_modified_or_deleted.yml:7]:
```suggestion
```

**Comment by nasbench** [comment]
Since most of these PRs are small fixes and are applying the same logic. Please next time do them all in one. To speed up the merging process :)

Thanks for your contribution once again :D

**Comment by TheLawsOfChaos** [comment]
@nasbench Sorry about that! I'm going through and adding stuff to my Jira board to work on converting it for my SIEM. I'll probably save them and submit them daily from now on!

**Comment by nasbench** [comment]
> @nasbench Sorry about that! I'm going through and adding stuff to my Jira board to work on converting it for my SIEM. I'll probably save them and submit them daily from now on!

No worries mate. It's just a suggestion but choose whatever process feels easier for you. I'm here to review :D 

Thanks for your contribution. Always happy to see people contributing in whatever way.

**Review by nasbench — APPROVED** [review]

**Pull request merged**